src/map/packets/char_abilities.cpp

```diff
@@ -34,7 +34,7 @@ CCharAbilitiesPacket::CCharAbilitiesPacket(CCharEntity* PChar)
 	this->size = 0x72;
 
 	memcpy(data+(0x04), PChar->m_WeaponSkills, 32);
-	memcpy(data+(0x46), PChar->m_Abilities,    62);
+	memcpy(data+(0x44), PChar->m_Abilities,    62);
```

Comment:
Nice find!

For anyone (like me) coming here and seeing this 62 and thinking something is up, it's meant to be 62 and not 64:
https://github.com/project-topaz/topaz/blob/release/src/map/entities/charentity.h#L196

Reply:
[Windower packet library has the following to say regarding these packets](https://github.com/Windower/Lua/blob/dev/addons/libs/packets/fields.lua#L3223):
> --0x0AA, 0x0AC, and 0x0AE are all bitfields where the lsb indicates whether you have index 0 of the related resource.

PVLV has the fields starting at 0x46, but Zeromus may have just copied us.

This solution pulls all the bits back by two bytes, which seems to make all the abilities line up with client expectations. But I'm not _100% positive_ if this is "proper" or if we should be messing with a LSB instead for certain action categories to deal with whatever wonky indexing the client is doing with the DATs.

@TeoTwawki might know more about DAT-indexing once the ID hits the client. I do know that all the IDs put in this PR align with what the server sends out on retail. But I don't know if our offset has just always been wrong (either an index shift, or intentionally so to align 0-based ID bits to the packet), or if it _should_ still be at 0x46 _with_ these IDs and we're missing a flag in the packet telling it which index to use. This signification may be related to the #663 family of issues; as while not part of the original quote in that issue, Thorny expounded further:
> it isn't just going to be a wrong table pull, their mapping is wrong but there are also some kind of bitflags to tell the game which table it's pulling from and they aren't documented anywhere

Reply:
> TeoTwawki might know more about DAT-indexing once the ID hits the client. I do know that all the IDs put in this PR align with what the server sends out on retail. 

@ibm2431 back when darkstar was a young project the offset used to make it match a dat we were grabbing the IDs from at that time, however I believe that SE has drastically changed that dat since that time and I fidn nothing matching the 16 offset now. Some other things that will not be affected by this PR are offset _elsewhere_ to match dats currently. The only reason I know of to do so is easy client extraction.